### PR TITLE
[Patch] Fixes `/metrics` endpoints with `serve-gunicorn`

### LINKED ISSUE #1662: `/metrics` doesn't get exposed while running with bentoml `serve-gunicorn`
**Describe the bug**

when running `bentoml serve-gunicorn` metrics doesn't get exposed as expected.

**To Reproduce**

1. Serve any current `BentoService` that you have (if none run `make torch-pipe` from [here](https://github.com/aarnphm/distributed-deployment-ml))
2. `docker run --gpus all -p 5000:5000 bento-torch-gpu:latest`
3. `curl -v localhost:5000/metrics`

**Expected behavior**

Terminal should returns prometheus format metrics, similar to when run `bentoml serve`

**Screenshots/Logs**
when run with `bentoml serve ...` (expected and correct behaviour)
image

when running in docker container (which is `bentoml serve-gunicorn`)
image
image

**Environment:**
 - OS: archlinux 5.12.8-arch1-1
 - Python Version: 3.8.8
 - BentoML Version: `master` branch 

**Additional context**

### PR DESCRIPTION
**Pull request by aarnphm**
## Description

Fix #1662. changed import order for ``prometheus_client`` in `yatai_service.py`

## Motivation and Context

GunicornBentoAPIServer should have used this `CollectorRegistry` https://github.com/bentoml/BentoML/blob/f9f9fbcbcb6e12a02cb005432cc5283529d81878/bentoml/server/instruments.py#L33 instead of creating an empty Registry https://github.com/bentoml/BentoML/blob/f9f9fbcbcb6e12a02cb005432cc5283529d81878/bentoml/server/gunicorn_server.py#L53 since no metrics are registered in that one.

## How Has This Been Tested?

``./ci/test_project.sh.cmd tests/integration/projects/general --pdb``

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [x] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [x] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [x] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/marshal/marshal.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def __init__(
                 labelnames=['endpoint'],
             )
             self.metrics_request_duration = Histogram(
-                name=service_name + '_mb_requestmb_duration_seconds',
+                name=service_name + '_mb_request_duration_seconds',
```
Comment (aarnphm) [inline bentoml/marshal/marshal.py:72]: typo.

File: bentoml/server/gunicorn_server.py [hunk 2]
```diff
@@ -45,13 +45,13 @@ class GunicornBentoAPIServer(BentoAPIServer):
         def metrics_view_func(self):
             from prometheus_client import (
                 CONTENT_TYPE_LATEST,
-                CollectorRegistry,
                 generate_latest,
                 multiprocess,
             )
 
-            registry = CollectorRegistry()
+            registry = self.app.wsgi_app.collector_registry
```
Comment (aarnphm) [inline bentoml/server/gunicorn_server.py:52]: This is assuming that our WSGI app only has `InstrumentMiddleware`. Are there a way for us to get the correct WSGI app if along the line we have more middleware for model server?

File: bentoml/yatai/yatai_service.py [hunk 3]
```diff
@@ -112,6 +111,7 @@ def start_yatai_service_grpc_server(
         PromServerInterceptor,
         ServiceLatencyInterceptor,
     )
+    from prometheus_client import start_http_server
```
Comment (aarnphm) [inline bentoml/yatai/yatai_service.py:114]: import order @ssheng suggested.

**Review by aarnphm — COMMENTED** [review]
@parano @yubozhao This should fix our `/metrics` endpoints.

File: bentoml/server/gunicorn_server.py (hunk 2 above)
Comment (aarnphm) [reply bentoml/server/gunicorn_server.py:52]: Maybe in the future implements a DispatcherMiddleware by path if we choose to have more middleware.

**Comment by bojiang** [comment]
https://github.com/bentoml/BentoML/blob/71f32494eb829b06ef90e7239e28a706f3da6501/tests/integration/projects/general/tests/test_meta.py#L10

If you want to add tests for `/metrics`, here is a suitable place.

**Comment by aarnphm** [comment]
> https://github.com/bentoml/BentoML/blob/71f32494eb829b06ef90e7239e28a706f3da6501/tests/integration/projects/general/tests/test_meta.py#L10
> 
> If you want to add tests for `/metrics`, here is a suitable place.

fixed it

File: tests/integration/projects/general/tests/test_meta.py [hunk 4]
```diff
@@ -68,3 +69,15 @@ def has_cors_headers(headers):
         assert_status=200,
         assert_headers=has_cors_headers,
     )
+
+
+@pytest.mark.parametrize(
+    "metrics",
+    ['_mb_request_duration_seconds_count', '_request_duration_seconds_bucket'],
```
Comment (parano) [inline tests/integration/projects/general/tests/test_meta.py:76]: these metrics will only show up after receiving the first inference request right? That seems to be the behavior I observed on Mac OS. Could that be the reason the API integration tests fail on Mac OS? perhaps we can send a prediction request before checking the "/metrics" endpoint in this test?

Comment (aarnphm) [reply tests/integration/projects/general/tests/test_meta.py:76]: the ``_mb`` metrics only shows up if the server is enable microbatching.

Comment (bojiang) [reply tests/integration/projects/general/tests/test_meta.py:76]: @aarnphm https://github.community/t/why-is-docker-not-installed-on-macos/17017/7

**Pull request merged**